src/Shared/Components/CICDHistory/DeploymentDetailSteps.tsx

```diff
@@ -0,0 +1,165 @@
+    const { appId, envId, triggerId } = useParams<{ appId: string; envId?: string; triggerId?: string }>()
+    const [deploymentListLoader, setDeploymentListLoader] = useState<boolean>(
+        deploymentStatus?.toUpperCase() !== TIMELINE_STATUS.ABORTED,
+    )
+    const isVirtualEnv = useRef(isVirtualEnvironment)
+    const processedData = isVirtualEnv.current
+        ? processVirtualEnvironmentDeploymentData()
+        : processDeploymentStatusDetailsData()
+    const [deploymentStatusDetailsBreakdownData, setDeploymentStatusDetailsBreakdownData] =
+        useState<DeploymentStatusDetailsBreakdownDataType>(processedData)
+
+    const clearDeploymentStatusTimer = (): void => {
+        if (deploymentStatusTimer) {
+            clearTimeout(deploymentStatusTimer)
+        }
+    }
+
+    const getDeploymentDetailStepsData = (): void => {
+        getDeploymentStatusDetail(appId, envId, true, triggerId, isHelmApps, installedAppVersionHistoryId)
+            .then((deploymentStatusDetailRes) => {
+                if (deploymentStatus !== 'Aborted') {
+                    // eslint-disable-next-line no-use-before-define
+                    processDeploymentStatusData(deploymentStatusDetailRes.result)
+                }
+            })
+            .catch(() => {
+                setDeploymentListLoader(false)
+            })
+            .finally(() => {
+                setDeploymentListLoader(false)
+            })
+    }
+
+    useEffect(() => {
+        if (deploymentAppType === DeploymentAppTypes.HELM) {
+            history.replace(`${url.replace('deployment-steps', 'source-code')}`)
+        }
+        if (isGitops) {
+            getDeploymentDetailStepsData()
+        }
+
+        return (): void => {
+            clearDeploymentStatusTimer()
+        }
+    }, [installedAppVersionHistoryId])
+
+    useEffect(() => {
+        isVirtualEnv.current = isVirtualEnvironment
+    }, [isVirtualEnvironment])
+
+    const processDeploymentStatusData = (deploymentStatusDetailRes: DeploymentStatusDetailsType): void => {
+        const processedDeploymentStatusDetailsData = isVirtualEnv.current
+            ? processVirtualEnvironmentDeploymentData(deploymentStatusDetailRes)
+            : processDeploymentStatusDetailsData(deploymentStatusDetailRes)
+        clearDeploymentStatusTimer()
+        // If deployment status is in progress then fetch data in every 10 seconds
+
+        // eslint-disable-next-line dot-notation
+        if (processedDeploymentStatusDetailsData['deploymentStatus'] === DEPLOYMENT_STATUS.INPROGRESS) {
+            deploymentStatusTimer = setTimeout(() => {
+                getDeploymentDetailStepsData()
+            }, 10000)
+        } else {
+            deploymentStatusTimer = setTimeout(() => {
+                getDeploymentDetailStepsData()
+            }, 30000)
+        }
+        setDeploymentStatusDetailsBreakdownData(processedDeploymentStatusDetailsData)
+    }
+
+    const redirectToDeploymentStatus = () => {
```

Comment:
we have converted ternary operation into if else nesting, if we are following this then we should leverage short circuiting the if condition and remove else block.